### PR TITLE
Pass write-service-info flag through to the flutter engine intents

### PR DESCRIPTION
**Pull request by jonahwilliams**
Part of the work for https://github.com/flutter/flutter/issues/41106

TBD: where does this file go?

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/engine/FlutterShellArgs.java [hunk 1]
```diff
@@ -47,6 +47,8 @@
   public static final String ARG_VERBOSE_LOGGING = "--verbose-logging";
   public static final String ARG_KEY_OBSERVATORY_PORT = "observatory-port";
   public static final String ARG_OBSERVATORY_PORT = "--observatory-port=";
+  public static final String ARG_KEY_WRITE_SERVICE_INFO = "write-service-info";
```
Comment (dnfield) [inline shell/platform/android/io/flutter/embedding/engine/FlutterShellArgs.java]: Shouldn't this also get wired up in the old embedding class?

Comment (jonahwilliams) [reply shell/platform/android/io/flutter/embedding/engine/FlutterShellArgs.java]: TIL we have more than one embedding

Comment (mklim) [reply shell/platform/android/io/flutter/embedding/engine/FlutterShellArgs.java]: Also nit: this should have documentation.

https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required

**Comment by jonahwilliams** [comment]
Spoke with Dan about this offline. We're not really sure that this is going to improve the observatory situation on Android so I'm closing it for now

**Pull request closed without merging**